src/core/trtserver.h

```diff
@@ -195,23 +210,23 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_SharedMemoryBlockDelete(
 /// case the function should return success and set 'buffer' ==
 /// nullptr.
 ///
-/// If the function is called with 'byte_size' non-zero but an
-/// allocation is not possible for the requested 'memory_type', the
-/// function should return success and set 'buffer' == nullptr to
-/// indicate that an allocation in the requested 'memory_type' is not
-/// possible. In this case the function may be called again for the
-/// same 'tensor_name' but with a different 'memory_type'.
+/// If the function is called with 'byte_size' non-zero the function should
+/// allocate a contiguous buffer of the requested size. If possible the function
+/// should allocate the buffer in the requested 'memory_type' and
+/// 'memory_type_id', but the function is free to allocate the buffer in any
+/// memory. The function must return in 'actual_memory_type' and
+/// 'actual_memory_type_id' the memory where the buffer is allocated.
 ///
 /// The function should return a TRTSERVER_Error object if a failure
 /// occurs while attempting an allocation. If an error object is
-/// returned, or if 'buffer' == nullptr is returned on all attempts
-/// for a result tensor, the inference server will assume allocation
+/// returned for a result tensor, the inference server will assume allocation
 /// is not possible for the result buffer and will abort the inference
 /// request.
 typedef TRTSERVER_Error* (*TRTSERVER_ResponseAllocatorAllocFn_t)(
```

Comment:
Sorry I didn't notice this before but the order of args in this function are incorrect. To be consistent with other API (and also not to be so jumbled even within this call) should have object (allocator in this case), followed by inputs, followed by outputs. So (allocator, tensor_name, byte_size, memory_type, memory_type_id, userp, buffer, buffer_userp, actual_memory_type, actual_memory_type_id)

Reply:
Fixed